src/date-input/date-input.tsx

```diff
@@ -69,21 +73,36 @@ export const DateInput = ({
     );
 
     /**
-     * Add handler function as dependencies of useEffect
-     * Allows our effect below to always get latest state value
+     * Allows effect below to always get latest state value
      * Reference:
      * https://stackoverflow.com/questions/65125665/new-event-doesnt-have-latest-state-value-updated-by-previous-event
      */
-    useEventListener("mousedown", handleInitEventListener);
+
+    useEventListener("keydown", handleKeyDown, nodeRef.current);
+    useEventListener("mousedown", handleMouseDown, document);
 
     // =============================================================================
     // REF FUNCTIONS
     // =============================================================================
     const calendarRef = useRef(null);
+    const isMouted = useRef(false);
 
     // =============================================================================
     // EFFECTS
     // =============================================================================
+    useEffect(() => {
+        // avoid to run in first mounted
+        if (!isMouted.current) {
+            isMouted.current = true;
+            return;
+        }
+
+        if (!calendarOpen) {
+            handleBlur(startDate, endDate);
+            setIsError(false);
+        }
```

Comment:
Good question!

Actually the value haven't been updated into reducer yet.  

Flow/Scenario:
a. blank input value -> select value -> clickOutSide/Cancel/Done -> useEffect able to get the latest value
b. blank input value -> select value -> clickOutSide/Cancel/Done -> handleBlurContainer get the previous value since the reducer not been updated yet.